src/lib/__tests__/credentials.test.ts

```diff
@@ -0,0 +1,195 @@
+    });
+
+    // We assume that Alice has only one credential and it's a credential from Faber
+    const [firstCredential] = await poll(
+      () => aliceAgent.credentials.getCredentials(),
+      (credentials: CredentialRecord[]) => credentials.length < 1,
+      100
+    );
+
+    expect(firstCredential).toEqual(
+      expect.objectContaining({
+        createdAt: expect.any(Number),
+        id: expect.any(String),
+        offer: {
+          '@id': expect.any(String),
+          '@type': 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/issue-credential/1.0/offer-credential',
+          comment: 'some comment about credential',
+          credential_preview: {
+            type: 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/issue-credential/1.0/credential-preview',
+            attributes: [
+              {
+                name: 'name',
+                mimeType: 'text/plain',
+                value: 'John',
+              },
+              {
+                name: 'age',
+                mimeType: 'text/plain',
+                value: '99',
+              },
+            ],
+          },
+          'offers~attach': expect.any(Array),
+        },
+        tags: {},
+        type: 'CredentialRecord',
+        state: CredentialState.OfferReceived,
+      })
+    );
+  });
+});
+
+async function registerSchema(agent: Agent, schemaTemplate: SchemaTemplate): Promise<[SchemaId, Schema]> {
+  const [schemaId] = await agent.ledger.registerCredentialSchema(schemaTemplate);
+  console.log('schemaId', schemaId);
+  const [ledgerSchemaId, ledgerSchema] = await agent.ledger.getSchema(schemaId);
+  console.log('ledgerSchemaId, ledgerSchema', ledgerSchemaId, ledgerSchema);
+  return [ledgerSchemaId, ledgerSchema];
+}
+
+async function registerDefinition(agent: Agent, definitionTemplate: CredDefTemplate): Promise<[CredDefId, CredDef]> {
+  const [credDefId] = await agent.ledger.registerCredentialDefinition(definitionTemplate);
+  const [ledgerCredDefId, ledgerCredDef] = await agent.ledger.getCredentialDefinition(credDefId);
+  console.log('ledgerCredDefId, ledgerCredDef', ledgerCredDefId, ledgerCredDef);
+  return [ledgerCredDefId, ledgerCredDef];
+}
+
+async function ensurePublicDidIsOnLedger(agent: Agent, publicDid: Did) {
+  try {
+    console.log(`Ensure test DID ${publicDid} is written to ledger`);
+    const agentPublicDid = await agent.ledger.getPublicDid(publicDid);
+    console.log(`Ensure test DID ${publicDid} is written to ledger: Success`, agentPublicDid);
+  } catch (error) {
+    // Unfortunately, this won't prevent from the test suite running because of Jest runner runs all tests
+    // regardless thorwn errors. We're more explicit about the problem with this error handling.
+    throw new Error(`Test DID ${publicDid} is not written on ledger or ledger is not available.`);
+  }
+}
+
+async function makeConnection(agentA: Agent, agentB: Agent) {
+  const { connection: aliceConnectionAtAliceBob, invitation } = await agentA.connections.createConnection();
```

Comment:
Not related to this PR, but maybe we should make the return type of `createConnection` an array just like the `createSchema` and `createCredentialDefinition` types. As the return types are almost always destructured, an array would allow to use a custom name for the return variable name.

Typescript 4.0 now also supports labeled tuple elements: https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/#labeled-tuple-elements

Reply:
That's an interesting idea. I would be worried about confusion in comparison with `createSchema` and `createCredentialDefinition` where you get `id` as the first and `object` as the second. As I'm thinking about that I've never liked this `createConnection` API. We should make `invitation` part of connection and add `role` = `Inviter/Invitee` as attribute.